libraries/Microsoft.Bot.Builder/ConversationState.cs

```diff
@@ -19,6 +20,16 @@ public ConversationState(IStorage storage)
         {
         }
 
+        public ConversationState(IStorage storage, int maxKeyLength)
+            : base(storage, nameof(ConversationState))
+        {
+            // Note: There is no min check here, as different data stores have different
+            // requiements.
+            MaxKeyLength = maxKeyLength;
+        }
+
+        public int MaxKeyLength { get; } = 254;
```

Comment:
Wouldn't this be a breaking change for any current storage implementation using conversation state keys longer than 254?

Reply:
It would. Let's review:

1. Azure tables supports a [1KB Size](The partition key may be a string value up to 1 KB in size.) limit for partition keys. We don't support Azure Tables so that's not an issue.

2. Azure storage supports a [1024 character blob name](https://blogs.msdn.microsoft.com/jmstall/2014/06/12/azure-storage-naming-rules/)

3. CosmosDB has a 256kb limit.

So any customer using blob storage would, as you point out, see thing break.